src/v2/controllers/vreplicaset_controller/proof/helper_lemmas.rs

```diff
@@ -137,16 +137,20 @@ pub proof fn lemma_filtered_pods_set_equals_matching_pods(
         Cluster::each_object_in_etcd_is_weakly_well_formed()(s),
         Cluster::etcd_is_finite()(s),
         resp_msg_is_the_in_flight_list_resp_at_after_list_pods_step(vrs, controller_id, resp_msg)(s),
+        forall |obj: DynamicObjectView| #[trigger] matching_pod_entries(vrs, s.resources()).values().contains(obj) ==> PodView::unmarshal(obj).is_Ok(),
     ensures
         ({
             let resp_objs = resp_msg.content.get_list_response().res.unwrap();
             let filtered_pods = filter_pods(objects_to_pods(resp_objs).unwrap(), vrs);
             &&& filtered_pods.no_duplicates()
             &&& filtered_pods.len() == matching_pod_entries(vrs, s.resources()).len() == matching_pods(vrs, s.resources()).len()
-            &&& filtered_pods.map_values(|p: PodView| p.marshal()).to_set() == matching_pod_entries(vrs, s.resources()).values()
+            // .mk_map(PodView::unmarshal).values() == .map_values(|p: PodView| p.marshal())
+            &&& filtered_pods.to_set() == matching_pod_entries(vrs, s.resources()).values().mk_map(|obj: DynamicObjectView| PodView::unmarshal(obj).get_Ok_0()).values()
```

Comment:
It appears `mk_map().values()` is a way to work around Verus's lack of a 'set map'... (purely a comment).